hw/ip/pinmux/doc/_index.md

```diff
@@ -6,122 +6,383 @@ title: "Pinmux Technical Specification"
+### MIO and DIO Signal Categories
+
+The `pinmux` supports two different IO signal categories:
+Muxed IO (MIO) signals that are routed through the `pinmux` matrix, and dedicated IO (DIO) signals that bypass the `pinmux` matrix.
+This allows to accommodate IO signals that are timing critical or that must have a fixed IO mapping for another reason.
+Note that although DIO signals are not routed through the `pinmux` matrix, they are still connected to the retention logic and the wakeup detectors (see next section below).
+
+The number of available peripheral IOs and pads and their assignment to the MIO / DIO categories is done at design time as part of the top-level configuration.
+This configurability is achieved by representing inputs / outputs as packed arrays, in combination with the SystemVerilog parameters `NPeriphIn`, `NPeriphOut` and `NMioPads` and `NDioPads`.
 Note however that the register file is also affected by this configuration and needs to be regenerated for each design instance.
 
-The assignment of which peripheral output goes to the output data signal of which chip pin is programmable, in other words, modifiable by software at run time.
-Similarly the assignment of which peripheral input is driven by which chip input is programmable.
-By default, all peripheral inputs are tied to zero or one (this default is design-time programmable via the `InputDefault` parameter).
+It is assumed that all available pins that the `pinmux` connects to are bidirectional, controlled by logic within this module.
+By default, all muxed peripheral inputs are tied to zero.
 Further, all output enables are set to zero, which essentially causes all pads to be in high-Z state after reset.
+In addition to wiring programmability, each muxed peripheral input can be set constantly to 0 or 1, and each muxed chip output can be set constantly to 0, 1 or high-Z.
 
-In addition to wiring programmability, each peripheral input can be set constantly to 0 or 1, and each chip output can be set constantly to 0, 1 or high-Z.
-Additional features such as output keeper or inversion are not provided by the pinmux since they are implemented in the `padctrl` IP which operates at the chip-level.
-This separation of physical pad attributes and logical wiring makes it possible to place the pads into an always-on domain without restricting possible power-down modes of the pinmux.
-One could imagine this to be a useful feature if a deep sleep mode were implemented in future; chip outputs could hold their output values and not be affected by internal power loss.
+See the [muxing matrix]({{< relref "#muxing-matrix">}}) section for more details about the mux implementation.
 
+### Retention and Wakeup Features
 
-# Theory of Operations
+The retention logic allows to specify a certain behavior during deep sleep for each muxed and dedicated output.
```

Comment:
The retention logic allows --> The retention logic allows you

Reply:
rewritten to `allows SW to` since again, `you` sounds a bit odd in this context...